### PR TITLE
Change isSamePerson definition to be of equal email and job ID

### LINKED ISSUE #131: Fix DuplicatePerson error when reopening the app
Should be solved after #145 solved (2 persons with same name and email but diff jobid)

image

### LINKED ISSUE #132: Fix ViewCommand to only view 1 Person at a time
Should be solved after #145 solved

### LINKED ISSUE #145: Include isSameJob in isSamePerson and remove isSameName
Will solve #131, #132, #146

### LINKED ISSUE #146: EditCommand not Showing Duplicate Person
Will be solved after #145 solved

```
add name/John phone/98765432 email/john@example.com address/311, Clementi Ave 2, #02-25 cap/3.50/4.00 gender/male university/Nanyang Polytechnic graduationDate/05-2024 major/Computer Science jobId/173296 jobTitle/Software Engineer Intern tag/rejected tag/KIV

add name/John phone/98765432 email/john@example.com address/311, Clementi Ave 2, #02-25 cap/3.50/4.00 gender/male university/Nanyang Polytechnic graduationDate/05-2024 major/Computer Science jobId/17329 jobTitle/Software Engineer Intern tag/rejected tag/KIV

edit <INDEX> jobId/173296
```

### PR DESCRIPTION
**Pull request by JasonCP14**
Resolve #131
Resolve #132 
Resolve #145 
Resolve #146

### CODE REVIEW

**Review by lulucopter — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 1]
```diff
@@ -39,39 +39,27 @@ public void isSamePerson() {
...
-
-        // different name, all other attributes same -> returns false
-        editedAlice = new PersonBuilder(ALICE).withName(VALID_NAME_BOB).build();
-        assertFalse(ALICE.isSamePerson(editedAlice));
-
-        // name has trailing spaces, all other attributes same -> returns false
-        String nameWithTrailingSpaces = VALID_NAME_BOB + " ";
-        editedBob = new PersonBuilder(BOB).withName(nameWithTrailingSpaces).build();
-        assertFalse(BOB.isSamePerson(editedBob));
-
```
Comment (JasonCP14) [inline src/test/java/seedu/address/model/person/PersonTest.java:71]: Moved to `EmailTest.java` and `NameTest.java`

**Review by cliftonfelix — CHANGES_REQUESTED** [review]
LGTM

**Review by cliftonfelix — APPROVED** [review]
LGTM

**Pull request merged**